File: 11/Thao-Solved/fctrl.py
#!/usr/bin/env python3
"""Solution to SPOJ problem #11 - Factorial."""

import math


def count_zeros(n):
    zeros = 0
    i = 1
    while math.pow(5, i) <= n:
        zeros += math.floor(n / (math.pow(5, i)))
        i += 1
    return int(zeros)


def main():
    """
    For any positive integer n, count the number of trailing zeros at the end
    of n!.
    """
    t = int(input())
    for i in range(t):
        n = int(input())
        print(count_zeros(n))


if __name__ == "__main__":
    main()

File: FCTRL/Thao-Solved/fctrl.cpp
// https://www.spoj.com/problems/FCTRL/

#include <iostream>
#include <math.h>

using namespace std;

int count_zeros(int n)
{
	int zeros = 0;

	for (int i=1; pow(5, i) <= n; i++)
	{
		zeros += floor(n / pow(5, i));
	}

	return zeros;
}

int main()
{	
	int t, n;
	cin >> t;

	while (t>0)
	{
		cin >> n;
		cout << count_zeros(n) << "\n";
		--t;
	}

	return 0;
}
